decidim-core/app/models/decidim/searchable_rsrc.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "pg_search"
+
+module Decidim
+  # A Searchable Resource.
+  # This is a model to a PgSearch table that indexes all searchable resources.
+  # This table is used to perform textual searches.
+  #
+  # Main attributes are:
+  # - locale: One entry per locale is required, so each resource will be indexed once per locale.
+  # - content_a: The most relevant textual content.
+  # - content_b: The second most relevant textual content.
+  # - content_c: The third most relevant textual content.
+  # - content_d: The less relevant textual content.
+  # - datetime:  The timestamp that places this resource in the line of time. Used as second criteria (first is text relevance) for sorting.
+  #
+  class SearchableRsrc < ApplicationRecord
```

Comment:
Could you please rename this to `SearchableResource`?

Reply:
First of all, I should say that I totally agree that names in software are very important and should be auto-descriptive. But I also think `SearchableRsrc` is much more concise and readable than `SearchableResource` that makes it harder to read. If we use it as a convention why not have this abbreviation?

Reply:
We don't have any convention specifying "Rsrc" means "Resource". And I certainly don't find "Rsrc" easier to read. It's easier to write (only 4 characters, vs 8 characters for "Resource"), but not easier to read. With that same argument we could be writing "Dcdm, Prpsls, Prcss, Mtng" 😕

Reply:
Here it comes the convention part. Ok, it was a proposal, I'll do the refactor.

By the way, you follow the same policy for one liners. Don't you think that for one liners it is best abbreviations?

Reply:
@tramuntanal now I'm lost, what one liners? 

We try to use complete words for our method, variable and constant names everywhere. We *might* have missed something, but that would have been an oversight for our part and not intentional

Reply:
not related with this specific PR, I was just talking in general. Maybe I should have opened an issue to discuss the use of abbreviations in oneliners, sorry.

Reply:
@tramuntanal truth is we try to avoid abbreviations in general, including oneliners. We might have missed something but that has been the general rule. Skipping a few letters creates confusion and have never seen clear benefits on using them.

The most compelling reason I can think of is a quote I once heard: "Code is written to be read by humans, not computers".

Reply:
Yes, with this discussions I seem to go even against my own principles of coding, but it also happens to me that when I see a lot of dense text it takes more time for my brain to read (and maybe understand). In oneliners (similarly as in conventions) the context is so small that it should be self evident what an abbreviation mean.
It is like writing a list of elements in one paragraph or in an item list. A paragraph is not visual, a list with one item per line is much more readable.
When we have names like `Decidim::ParicipatoryProcesses::ParticipatoryProcess` in the middle of dense code I don't realize if it is a participatory process class or a proposal class until I read the end of the namespace. Probably this is why I was trying to use `Decidim::SearchableRsrc` `Decidim::SearchableResource`, it seems to me that my brain captures the name faster. It is also true that we can not use abbreviations for every new model as @mrcasals pointed out.